editor/src/components/canvas/canvas-strategies/strategies/reparent-helpers/reparent-strategy-parent-lookup.ts

```diff
@@ -147,15 +166,20 @@ function findValidTargetsUnderPoint(
   ]
 
   const possibleTargetParentsUnderPoint = mapDropNulls((target) => {
-    const conditionalChildWithEmptyBranch = MetadataUtils.getChildrenOrdered(
-      metadata,
-      elementPathTree,
-      target,
-    )
-      .map((c) => c.elementPath)
-      .find((path) => isConditionalWithEmptyActiveBranch(path, metadata))
-    if (conditionalChildWithEmptyBranch != null) {
-      return conditionalChildWithEmptyBranch
+    if (!EP.isStoryboardPath(target)) {
+      const subtree = getSubTree(elementPathTree, target)
```

Comment:
what's the cost of getSubTree? because we're going to call this for every element under the mouse for every frame during every Strategy interaction

Reply:
at a very quick glance, it's an O(n^2), maybe it would be nice to cache it, traversing the tree is gonna be needed one way or the other 🧐

Reply:
I don't think you need to get the subtree here, you can just traverse conditionals until the active branch is either another conditional or empty. And then it will be linear compared to the number of directly embedded conditionals